chain/stmgr/utils.go

```diff
@@ -240,7 +249,8 @@ func GetSectorsForWinningPoSt(ctx context.Context, pv ffiwrapper.Verifier, sm *S
 		return nil, xerrors.Errorf("generating winning post challenges: %w", err)
 	}
 
-	sectors, err := mas.LoadSectorsFromSet(&provingSectors, false)
+	// we don't need to filter here (and it's **very** slow)
```

Comment:
For context - loading those sectors with the filter as currently implemented was taking 4+ mins for some miners (with chain data on an NVMe)

```
goroutine 394544 [runnable]:
github.com/filecoin-project/go-bitfield/rle.(*rleIterator).NextRun(0xc11d2ecb20, 0xc0e5774001, 0x2, 0x0, 0x0)
	/home/magik6k/.opt/go/pkg/mod/github.com/filecoin-project/go-bitfield@v0.2.0/rle/rleplus_reader.go:41 +0x72
github.com/filecoin-project/go-bitfield/rle.(*RLE).RunIterator(0xc0d15ef2a8, 0xc0f0d70ff0, 0x16f219, 0x513c040, 0x0)
	/home/magik6k/.opt/go/pkg/mod/github.com/filecoin-project/go-bitfield@v0.2.0/rle/rleplus.go:50 +0x1fd
github.com/filecoin-project/go-bitfield.BitField.IsSet(0xc100b18a80, 0x5524, 0x6a80, 0x0, 0x0, 0x0, 0xc0f0d70fc0, 0xc0f0d70ff0, 0x16f219, 0x1, ...)
	/home/magik6k/.opt/go/pkg/mod/github.com/filecoin-project/go-bitfield@v0.2.0/bitfield.go:516 +0x94
github.com/filecoin-project/lotus/chain/actors/builtin/miner.(*state0).LoadSectorsFromSet.func1(0x16f219, 0x0, 0x0)
	/home/magik6k/github.com/filecoin-project/go-lotus/chain/actors/builtin/miner/v0.go:169 +0xde
github.com/filecoin-project/specs-actors/actors/util/adt.(*Array).ForEach.func1(0x16f219, 0xc11d2ec040, 0x0, 0x0)
	/home/magik6k/.opt/go/pkg/mod/github.com/filecoin-project/specs-actors@v0.9.10/actors/util/adt/array.go:90 +0x82
github.com/filecoin-project/go-amt-ipld/v2.(*Node).forEachAt(0xc0fdad5d80, 0x2f0db60, 0xc13aeba240, 0x7f08b51d7500, 0xc12526b8e0, 0x0, 0x0, 0x16f218, 0xc0d15ef910, 0x0, ...)
	/home/magik6k/.opt/go/pkg/mod/github.com/filecoin-project/go-amt-ipld/v2@v2.1.0/amt.go:274 +0x43b
github.com/filecoin-project/go-amt-ipld/v2.(*Node).forEachAt(0xc0fdad5600, 0x2f0db60, 0xc13aeba240, 0x7f08b51d7500, 0xc12526b8e0, 0x1, 0x0, 0x16f200, 0xc0d15ef910, 0xc0d15ef5e0, ...)
	/home/magik6k/.opt/go/pkg/mod/github.com/filecoin-project/go-amt-ipld/v2@v2.1.0/amt.go:306 +0x1f4
github.com/filecoin-project/go-amt-ipld/v2.(*Node).forEachAt(0xc0fdad5400, 0x2f0db60, 0xc13aeba240, 0x7f08b51d7500, 0xc12526b8e0, 0x2, 0x0, 0x16f200, 0xc0d15ef910, 0x0, ...)
	/home/magik6k/.opt/go/pkg/mod/github.com/filecoin-project/go-amt-ipld/v2@v2.1.0/amt.go:306 +0x1f4
github.com/filecoin-project/go-amt-ipld/v2.(*Node).forEachAt(0xc141a66980, 0x2f0db60, 0xc13aeba240, 0x7f08b51d7500, 0xc12526b8e0, 0x3, 0x0, 0x16f000, 0xc0d15ef910, 0x0, ...)
	/home/magik6k/.opt/go/pkg/mod/github.com/filecoin-project/go-amt-ipld/v2@v2.1.0/amt.go:306 +0x1f4
github.com/filecoin-project/go-amt-ipld/v2.(*Node).forEachAt(0xc108871100, 0x2f0db60, 0xc13aeba240, 0x7f08b51d7500, 0xc12526b8e0, 0x4, 0x0, 0x168000, 0xc0d15ef910, 0x0, ...)
	/home/magik6k/.opt/go/pkg/mod/github.com/filecoin-project/go-amt-ipld/v2@v2.1.0/amt.go:306 +0x1f4
github.com/filecoin-project/go-amt-ipld/v2.(*Node).forEachAt(0xc0ff1ddd80, 0x2f0db60, 0xc13aeba240, 0x7f08b51d7500, 0xc12526b8e0, 0x5, 0x0, 0x140000, 0xc0d15ef910, 0x0, ...)
	/home/magik6k/.opt/go/pkg/mod/github.com/filecoin-project/go-amt-ipld/v2@v2.1.0/amt.go:306 +0x1f4
github.com/filecoin-project/go-amt-ipld/v2.(*Node).forEachAt(0xc0f0d028d0, 0x2f0db60, 0xc13aeba240, 0x7f08b51d7500, 0xc12526b8e0, 0x6, 0x0, 0x0, 0xc0d15ef910, 0x0, ...)
	/home/magik6k/.opt/go/pkg/mod/github.com/filecoin-project/go-amt-ipld/v2@v2.1.0/amt.go:306 +0x1f4
github.com/filecoin-project/go-amt-ipld/v2.(*Root).ForEach(...)
	/home/magik6k/.opt/go/pkg/mod/github.com/filecoin-project/go-amt-ipld/v2@v2.1.0/amt.go:256
github.com/filecoin-project/specs-actors/actors/util/adt.(*Array).ForEach(0xc0f11994a0, 0x2ed71a0, 0xc0f11994c0, 0xc0d1661980, 0xc0f11994a0, 0x0)
	/home/magik6k/.opt/go/pkg/mod/github.com/filecoin-project/specs-actors@v0.9.10/actors/util/adt/array.go:81 +0xe6
github.com/filecoin-project/lotus/chain/actors/builtin/miner.(*state0).LoadSectorsFromSet(0xc0cf924000, 0xc0ec1a3480, 0xc13aeba200, 0x3, 0x4f8, 0xc0e2fed860, 0x20)
	/home/magik6k/github.com/filecoin-project/go-lotus/chain/actors/builtin/miner/v0.go:167 +0x115
github.com/filecoin-project/lotus/chain/stmgr.GetSectorsForWinningPoSt(0x2f0db60, 0xc13aeba240, 0x2f0fa20, 0x513a438, 0xc0d23b6800, 0xc0e129d4d0, 0x26, 0xc13c611436, 0x3, 0xc0e2fed860, ...)
	/home/magik6k/github.com/filecoin-project/go-lotus/chain/stmgr/utils.go:252 +0xe02
github.com/filecoin-project/lotus/chain.(*Syncer).VerifyWinningPoStProof(0xc0006ceea0, 0x2f0db60, 0xc13aeba240, 0xc13c630380, 0x28639, 0xc0cd9b0840, 0x60, 0x60, 0xc0e129d4d0, 0x26, ...)
	/home/magik6k/github.com/filecoin-project/go-lotus/chain/sync.go:1001 +0x539
github.com/filecoin-project/lotus/chain.(*Syncer).ValidateBlock.func10(0xc0d1211800, 0x780)
	/home/magik6k/github.com/filecoin-project/go-lotus/chain/sync.go:920 +0xa5
github.com/Gurpartap/async.Err.func1(0xc0e1907aa0, 0xc067fcf1d0, 0xc115e1d450)
	/home/magik6k/.opt/go/pkg/mod/github.com/!gurpartap/async@v0.0.0-20180927173644-4f7f499dd9ee/error.go:29 +0x53
created by github.com/Gurpartap/async.Err
	/home/magik6k/.opt/go/pkg/mod/github.com/!gurpartap/async@v0.0.0-20180927173644-4f7f499dd9ee/error.go:27 +0x8a
```

Reply:
This has been rewritten.